packages/mattermost-components/src/generic_modal/generic_modal.tsx

```diff
@@ -0,0 +1,521 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+import styled from 'styled-components';
+
+type Props = {
+    className?: string;
+    onExited: () => void;
+    modalHeaderText?: React.ReactNode;
+    show?: boolean;
+    handleCancel?: () => void;
+    handleConfirm?: () => void;
+    handleEnterKeyPress?: () => void;
+    confirmButtonText?: React.ReactNode;
+    confirmButtonClassName?: string;
+    cancelButtonText?: React.ReactNode;
+    cancelButtonClassName?: string;
+    isConfirmDisabled?: boolean;
+    id: string;
+    autoCloseOnCancelButton?: boolean;
+    autoCloseOnConfirmButton?: boolean;
+    enforceFocus?: boolean;
+    container?: React.ReactNode | React.ReactNodeArray;
+    ariaLabel?: string;
+    errorText?: string;
+    compassDesign?: boolean;
+    backdrop?: boolean;
+    backdropClassName?: string;
+};
+
+type State = {
+    show: boolean;
+}
+
+export class GenericModal extends React.PureComponent<Props, State> {
```

Comment:
Are there any major changes to this component, or was it mostly just a move plus putting the CSS into the file?

Reply:
No major changes.